packages/composite-checkout/src/components/billing-fields.js

```diff
@@ -0,0 +1,221 @@
+} ) {
+	const [ items ] = useCheckoutLineItems();
+	const localize = useLocalize();
+	if ( summary && isComplete ) {
+		return <BillingFormSummary paymentData={ paymentData } />;
+	}
+	if ( ! isActive ) {
+		return null;
+	}
+	const {
+		billingName = '',
+		billingAddress = '',
+		billingCity = '',
+		billingState = '',
+		billingZipCode = '',
+		billingCountry = '',
+		billingPhoneNumber = '',
+		billingVatIdNumber = '',
+		billingNameError = null,
+		billingAddressError = null,
+		billingCityError = null,
+		billingStateError = null,
+		billingZipCodeError = null,
+		billingCountryError = null,
+		billingPhoneNumberError = null,
+		billingVatIdError = null,
+	} = paymentData || {};
+
+	const nameOnChange = value => setPaymentData( { ...( paymentData || {} ), billingName: value } );
+	const addressOnChange = value =>
+		setPaymentData( { ...( paymentData || {} ), billingAddress: value } );
+	const cityOnChange = value => setPaymentData( { ...( paymentData || {} ), billingCity: value } );
+	const stateOnChange = value =>
+		setPaymentData( { ...( paymentData || {} ), billingState: value } );
+	const zipCodeOnChange = value =>
+		setPaymentData( { ...( paymentData || {} ), billingZipCode: value } );
+	const countryOnChange = value =>
+		setPaymentData( { ...( paymentData || {} ), billingCountry: value } );
+	const phoneNumberOnChange = value =>
+		setPaymentData( { ...( paymentData || {} ), billingPhoneNumber: value } );
+	const vatIdOnChange = value =>
+		setPaymentData( { ...( paymentData || {} ), billingVatIdNumber: value } );
+
+	return (
+		<BillingFormFields>
+			{ isDomainRelatedField( items ) && (
+				<React.Fragment>
+					<FormField
+						id="billing-name"
+						type="Text"
+						label={ localize( 'Name' ) }
+						error={ billingNameError }
+						errorMessage={ localize( 'This is a required field' ) }
+						value={ billingName }
+						onChange={ nameOnChange }
+					/>
+
+					<FormField
+						id="billing-address"
+						type="Text"
+						label={ localize( 'Address' ) }
+						error={ billingAddressError }
+						errorMessage={ localize( 'This is a required field' ) }
+						value={ billingAddress }
+						onChange={ addressOnChange }
+					/>
+
+					<FieldRow gap="4%" columnWidths="48% 48%">
+						<Field
+							id="billing-city"
+							type="Text"
+							label={ localize( 'City' ) }
+							error={ billingCityError }
+							errorMessage={ localize( 'This is a required field' ) }
+							value={ billingCity }
+							onChange={ cityOnChange }
+						/>
+
+						<Field
+							id="billing-state"
+							type="Text"
+							label={ localize( 'State' ) }
+							error={ billingStateError }
+							errorMessage={ localize( 'This is a required field' ) }
+							value={ billingState }
+							onChange={ stateOnChange }
+						/>
+					</FieldRow>
+				</React.Fragment>
+			) }
+
+			<FieldRow gap="4%" columnWidths="48% 48%">
+				<Field
+					id="billing-zip-code"
+					type="Text"
+					label={ localize( 'Zip code' ) }
```

Comment:
This should probably be "Postal code", right? Since I think "zip code" is just a US name? I could be wrong.

Reply:
It would depend whether we're in the US or not. I updated the component to serve up a Zip or Post Code field. I've stubbed in the logic for now but we'll need to detect location in order to show the right field. (I did this for State and Province too).